src/foundations/icon/Icon.props.ts

```diff
@@ -1,23 +1,29 @@
-import { TIconColors, TIconGlyph, TIconSize } from './Icon.types';
+import { TIconColor, TIconGlyph, TIconSize } from './Icon.types';
 
-export type PIcon = {
-    /**
-     * the icon-glyph that is being rendered
-     * @default 'mattermost'
-     * */
-    glyph?: TIconGlyph;
+type PIcon = {
     /**
      * the size the icon is rendered with
      * @default 20
-     * */
+     */
     size?: TIconSize;
+    /**
+     * the icon-glyph that is being rendered
+     * @default 'mattermost'
+     */
```

Comment:
nit: since we don't pass defaults for values like icons and images do we wan't this one?